src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/UrlSegment.cs

```diff
@@ -1,13 +1,15 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Http.Extensions;
+
 namespace Microsoft.AspNetCore.Rewrite.Internal.PatternSegments
 {
     public class UrlSegment : PatternSegment
     {
         public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
         {
-            return context.HttpContext.Request.Path;
+            return context.GlobalRule ? context.HttpContext.Request.GetEncodedUrl() : (string)context.HttpContext.Request.Path;
```

Comment:
@mikaelm12 We need to check if this gives the expected escaping compared with the native module.